modules/account-lib/src/coin/near/transaction.ts

```diff
@@ -126,30 +141,48 @@ export class Transaction extends BaseTransaction {
     this.loadInputsAndOutputs();
   }
 
+  /**
+   * set transaction type by staking contract method names.
+   * @param methodName method name to match and set the transaction type
+   */
+  private setTypeByStakingMethod(methodName: string): void {
+    const stakingContractTypes = {
+      [StakingContractMethodNames.DepositAndStake]: this.setTransactionType(TransactionType.StakingActivate),
+    };
+    stakingContractTypes[methodName];
+  }
+
   /**
    * Build input and output field for this transaction
    *
    */
-
   loadInputsAndOutputs(): void {
-    if (this._nearTransaction.actions.length != 1) {
+    if (this._nearTransaction.actions.length !== 1) {
       throw new InvalidTransactionError('too many actions in raw transaction');
     }
 
-    if (this._nearTransaction.actions[0].enum === 'transfer') {
-      this.setTransactionType(TransactionType.Send);
-    } else {
-      throw new InvalidTransactionError('unsupported action in raw transaction');
+    const action = this._nearTransaction.actions[0];
+
+    switch (action.enum) {
+      case 'transfer':
+        this.setTransactionType(TransactionType.Send);
+        break;
+      case 'functionCall':
+        const methodName = action.functionCall.methodName;
+        if (methodName in StakingContractMethodNames) {
```

Comment:
I think this is wrong. If the method is in the list of valid methods shouldn't throw an error. "In" is used with objects, I'm not sure that it works with enums.

Reply:
Its wrong, nice catch

Reply:
Its doing a fake falsy actually, making that it "works" but its a bug.